public/js/main.js

```diff
@@ -3,121 +3,192 @@
  * @license   MIT
  */
 "use strict";
-$(document).ready(function(){
-    $('#get-url-btn').click(function(){
-        clearWarnings();
-        if (passwordConfirmed()) {
-            var password = $('#password-field').val();
-            var views = $('input[name=views]').val();
-            var ttl = ($('input[name=hours]').val() * 3600) + ($('input[name=days]').val() * 3600 * 24);
-            $.post('/backend.php', {password: password, ttl: ttl, views: views}, function(secretKey){
-                showUrl(secretKey);
-            });
-        }
-    });
-    
-    $('#views-select li a').click(function(event){
-        event.preventDefault();
-        var views = $(this).data('views');
-        var btnTxt = 'Valid for '+views+' view';
-        if(views > 1) {
-            btnTxt += 's';
+
+var phpsst = new Vue({
+    el: '#phpsst',
+    data: {
+        errorMsg: '',
+        passwordDisplay: '',
+        secretUrl: '',
+        passwordField: '',
+        passwordConfirmField: '',
+        views: 1,
+        hours: 0,
+        days: 1,
+        page: 'enter-details',
+        selectViewsBtnText: 'Valid for 1 view',
+        selectDaysBtnText: 'Valid for 1 day',
+        selectHoursBtnText: 'and 0 hour',
+        selectDays: [],
+        selectHours: [],
+        selectViews: []
+    },
+    mounted: function () {
+        var i = 0;
+        for (i = 0; i <= 10; i++) {
+            this.selectDays.push(this.getDaysLabel(i));
         }
-        $('#select-valid-btn-txt').text(btnTxt);
-        $('input[name=views]').val(views);
-    });
-    
-    $('#days-select li a').click(function(event){
-        event.preventDefault();
-        var days = $(this).data('days');
-        var btnTxt = 'Valid for '+days+' day';
-        if(days > 1) {
-            btnTxt += 's';
+        for (i = 0; i <= 23; i++) {
+            this.selectHours.push(this.getHoursLabel(i));
         }
-        $('#select-days-btn-txt').text(btnTxt);
-        $('input[name=days]').val(days);
-    });
-    
-    $('#hours-select li a').click(function(event){
-        event.preventDefault();
-        var hours = $(this).data('hours');
-        var btnTxt = 'and '+hours+' hour';
-        if(hours > 1) {
-            btnTxt += 's';
+        for (i = 1; i <= 10; i++) {
+            this.selectViews[i] = this.getViewsLabel(i);
         }
-        $('#select-hours-btn-txt').text(btnTxt);
-        $('input[name=hours]').val(hours);
-    });
 
-    $('.page-jumper').click(function() {
-        resetInputFields();
-        $('#alert-container').html('');
-        showPage($(this).data('page'));
-    });
+        window.onhashchange = function () {
```

Comment:
The issue is that you cannot add v-on:hashchange="myFunction" to the body - since Vue does not allow one to bind a Vue instance to the body or html tags. And since the onhashchange-event is only triggered on the body tagg I don't really see a way around it.